### DISCUSSION TITLE
Pasting a template

### DISCUSSION THREAD

**Discussion by latvia234** [vincode-io/Zavala#187] [Ideas]
It would be helpful if text expanders could paste templates in Zavala. (I'm using [Keyboard Maestro](https://www.keyboardmaestro.com/main/) as a text expander.) [Here](https://workflowy.com/templates/) you can find some templates for testing. I'm using [this template](https://workflowy.com/template/1-minute-minutes/).

Current result when pasting a template via cmd+v in edit mode:
![Schermafbeelding 2022-09-12 om 08 14 33](https://user-images.githubusercontent.com/86680397/189585485-ea5d487e-99c3-44be-8e4a-fdd1e64c47ee.png)
Current result when pasting a template via cmd+v in view mode:
![Schermafbeelding 2022-09-12 om 08 15 41](https://user-images.githubusercontent.com/86680397/189585634-5c397560-1ce6-4e95-968d-0e301278251f.png)

Expected result (using [Dynalist](https://dynalist.io)):
![Schermafbeelding 2022-09-12 om 08 18 19](https://user-images.githubusercontent.com/86680397/189585994-c6ef06f6-d175-4616-b4b4-c7520c95bf2b.png)

**Comment by vincode-io** [comment]
Did you recreate this Template in KeyboardMaestro [manually](https://www.keyboardmaestro.com/documentation/6/macroexamples.html#macroexamples_inserttext)? I'm not sure how WorkFlowy Templates could be used otherwise.

**Comment by latvia234** [comment]
Yes I did recreate it manually and pasted it in a macro inside KeyboardMaestro. But you can get the same result if you just copy it from Workflowy into Text editor. And copy it from Text editor into Zavala.

**Reply by vincode-io** [reply]
I think I see what you are saying now. When pasting while not editing in a row, we should use tabs in the pasted text to understand when a row is a child row.

If you agree, I'll write this up as a bug and put it into a future bug fix release.

**Reply by latvia234** [reply]
> I think I see what you are saying now. When pasting while not editing in a row, we should use tabs in the pasted text to understand when a row is a child row.

Yes that would be great. When using text expanders like Keyboard Maestro and Alfred for pasting templates this would be useful.

> If you agree, I'll write this up as a bug and put it into a future bug fix release.

Sure 👍

**Comment by vincode-io** [comment]
The issue to track this is #189. I'll put it in a future bug fix release.

**Reply by latvia234** [reply]
Will issue #189 also fix dragging from OmniOutliner (or another outline) into Zavala?

https://user-images.githubusercontent.com/86680397/191105767-f26cfeb0-d18a-4307-a198-422a7af8946c.mov

**Comment by vincode-io** [comment]
I don’t know. I have no idea what the pasteboard looks like when OO loads it. If this doesn’t fix it, I address it in another issue.
